youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,243 @@
+            '4500': [1920, 1080]
+        }
+        _PATTERN = r'(rrtv-([\d\,]+)-)'
+
+        def get_format_info(tbr):
+            br = int_or_none(tbr)
+            if len(formats) == 1 and not br:
+                br = formats[0].get('tbr')
+
+            for f in formats:
+                if f.get('tbr'):
+                    if (br - br / 100 * 10) <= f['tbr'] <= (br + br / 100 * 10):
+                        return [
+                            f.get('width'),
+                            f.get('height'),
+                            f['tbr']
+                        ]
+            return [None, None, None]
+
+        mobj = re.search(_PATTERN, mp4.get('mp4') or '')
+        if not mobj:
+            return None
+        pattern = mobj.group(1)
+
+        qualities = re.search(_PATTERN, mp4.get('manifest') or '')
+        if qualities:
+            qualities = qualities.group(2)
+            qualities = qualities.split(',') if qualities else ['.']
+            qualities = [i for i in qualities if i]
+        else:
+            qualities = [mobj.group(2)]
+
+        http_formats = []
+        for q in qualities:
+            w, h, t = get_format_info(q)
+            http_formats.append({
+                'url': mp4['mp4'].replace(pattern, 'rrtv-%s-' % q),
+                'width': w or _QUALITY[q][0],
+                'height': h or _QUALITY[q][1],
+                'tbr': t or int(q),
+                'protocol': 'https',
+                'format_id': 'https-%s' % q,
+            })
+        return http_formats
+
+    def _real_extract(self, url):
+        u = re.match(self._VALID_URL, url)
+        self.IE_NAME = u.group('iename') if u.group('iename') else 'gedi'
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        player_data = re.findall(
+            r'PlayerFactory\.setParam\(\'(?P<type>.+?)\',\s*\'(?P<name>.+?)\',\s*\'(?P<val>.+?)\'\);',
+            webpage)
+
+        formats = []
+        audio_fmts = []
+        title = None
+        thumb = None
+        mp4 = {}
+
+        for t, n, v in player_data:
+            if t == 'format':
+                if n == 'mp4':
+                    mp4.update({'mp4': v})
+                if n == 'video-hls-vod-ak':
+                    mp4.update({'manifest': v})
+                    formats.extend(self._extract_akamai_formats(
+                        v, video_id, {'http': 'media.gedidigital.it'}))
+                if n == 'audio-hls-vod':
+                    audio_fmts.extend(self._extract_m3u8_formats(
+                        v, video_id, 'm4a', m3u8_id='audio-hls', fatal=False))
+                if n == 'audio-rrtv-mp3':
+                    audio_fmts.append({
+                        'format_id': 'mp3',
+                        'url': v,
+                        'tbr': 128,
+                        'ext': 'mp3',
+                        'vcodec': 'none',
+                        'acodec': 'mp3',
+                    })
```

Comment:
- extract all formats.
- try not complicating things, try to avoid using hardcoded values.